### PR TITLE
[Quantum] Add "Microsoft.AzureQuantum-" prefix to deployment name

### PR DESCRIPTION
**Pull request by warren-jones**
To follow our telemetry naming convention and standardize queries, the ARM deployment name should begin with “Microsoft.AzureQuantum-“ when creating a quantum workspace.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

### CODE REVIEW

**Comment by yonzhan** [comment]
Quantum

File: src/quantum/azext_quantum/tests/latest/test_quantum_workspace.py [hunk 1]
```diff
@@ -58,6 +58,7 @@ def test_workspace_create_destroy(self):
         test_storage_account = get_test_workspace_storage()
         test_storage_account_grs = get_test_workspace_storage_grs()
         test_provider_sku_list = get_test_workspace_provider_sku_list()
+        #test_deployment_name = f"Microsoft.AzureQuantum-{test_workspace_temp}"
```
Comment (vxfield) [inline src/quantum/azext_quantum/tests/latest/test_quantum_workspace.py:61]: remove this comment?

File: src/quantum/azext_quantum/tests/latest/test_quantum_workspace.py [hunk 2]
```diff
@@ -77,7 +78,7 @@ def test_workspace_create_destroy(self):
 
             # create
             self.cmd(f'az quantum workspace create -g {test_resource_group} -w {test_workspace_temp} -l {test_location} -a {test_storage_account} -r {test_provider_sku_list} -o json', checks=[
-            self.check("name", test_workspace_temp),
+            self.check("name", DEPLOYMENT_NAME_PREFIX + test_workspace_temp),
             # >>>>>self.check("provisioningState", "Succeeded")  # Status is "Succeeded" since we are linking the storage account this time.
```
Comment (vxfield) [inline src/quantum/azext_quantum/tests/latest/test_quantum_workspace.py:82]: remove this comment?

**Review by vxfield — APPROVED** [review]
Approved with some feedback to remove to unused commented lines.

**Review by vxfield — APPROVED** [review]
Approved with some feedback to remove commented lines

File: src/quantum/azext_quantum/tests/latest/test_quantum_workspace.py [hunk 3]
```diff
@@ -92,7 +93,7 @@ def test_workspace_create_destroy(self):
 
             # create
             self.cmd(f'az quantum workspace create -g {test_resource_group} -w {test_workspace_temp} -l {test_location} -a {test_storage_account_grs} -r {test_provider_sku_list} -o json', checks=[
-            self.check("name", test_workspace_temp),
+            self.check("name", DEPLOYMENT_NAME_PREFIX + test_workspace_temp),
             # >>>>>self.check("provisioningState", "Succeeded")  # Status is "Succeeded" since we are linking the storage account this time.
```
[the remark this answers is not in the document]
Comment (warren-jones) [reply src/quantum/azext_quantum/tests/latest/test_quantum_workspace.py]: Removed it

File: src/quantum/azext_quantum/tests/latest/test_quantum_workspace.py (hunk 2 above)
Comment (warren-jones) [reply src/quantum/azext_quantum/tests/latest/test_quantum_workspace.py]: Removed it

File: src/quantum/azext_quantum/operations/workspace.py [hunk 4]
```diff
@@ -253,7 +254,7 @@ def create(cmd, resource_group_name=None, workspace_name=None, location=None, st
 
     deployment_async_operation = arm_client.deployments.begin_create_or_update(
         info.resource_group,
-        workspace_name,     # Note: This is actually specifying a the deployment name, but workspace_name is used here in test_quantum_workspace.py
+        DEPLOYMENT_NAME_PREFIX + workspace_name,
```
Comment (george-moussa) [inline src/quantum/azext_quantum/operations/workspace.py:257]: is there a max size to the deployment name?

**Review by george-moussa — APPROVED** [review]

File: src/quantum/azext_quantum/operations/workspace.py (hunk 4 above)
Comment (warren-jones) [reply src/quantum/azext_quantum/operations/workspace.py]: Added code to truncate long deployment names to 64 characters.

**Review by necusjz — APPROVED** [review]

**Pull request merged**